### PR TITLE
utils.crypto: Fix two errors in hash_file

### PR DESCRIPTION
**Pull request by luckyh**
Fix the following two errors in hash_file:

* A wrong indentation caused a dead loop
* Forgot returning `None` when the given hash algorithm is unknown

Signed-off-by: Xu Han <xuhan@redhat.com>

### CODE REVIEW

**Comment by luckyh** [comment]
@clebergnu @apahim please help review this ASAP, thanks.

File: avocado/utils/crypto.py [hunk 1]
```diff
@@ -66,11 +66,11 @@ def hash_file(filename, size=None, algorithm="md5"):
         while size > 0:
```
Comment (apahim) [inline avocado/utils/crypto.py]: The try/except clause above is broken. Mind fixing (in a separate commit)? Just add a `return None` right after the `logging.error(Unknown...`

File: avocado/utils/crypto.py [hunk 2]
```diff
@@ -66,11 +66,11 @@ def hash_file(filename, size=None, algorithm="md5"):
...
             if chunksize > size:
                 chunksize = size
-                data = file_to_hash.read(chunksize)
-                if len(data) == 0:
-                    logging.debug("Nothing left to read but size=%d", size)
-                    break
-                hash_obj.update(data)
-                size -= len(data)
```
Comment (apahim) [inline avocado/utils/crypto.py]: ok, checking

File: avocado/utils/crypto.py (hunk 1 above)
Comment (luckyh) [reply avocado/utils/crypto.py]: Ah, sure, will fix it.

**Pull request merged**